### ISSUE TITLE
Partial support for Ruby?

### ISSUE THREAD

**Issue by jensimmons** [Fyrd/caniuse#7136]
https://caniuse.com/ruby
<img width="1373" alt="Screenshot 2024-08-29 at 2 28 53 PM" src="https://github.com/user-attachments/assets/2eea1aef-91b9-47fa-ab9a-eac4c6535f40">

>Partial support refers to only supporting basic ruby, may still be missing writing-mode, Complex ruby and CSS3 Ruby.

I'm wondering if this is accurate anymore. Without specific details of what's missing to be considered "full support", it's unclear if Safari or Chrome have full support.

**Comment by jensimmons** [comment]
Safari Technology Preview 202 just added support for ruby-align and ruby-position, strengthening our support for CSS that lets authors refine ruby layout. But I don't think that's what this page is about. This seems to be about the core HTML support, with vertical writing mode. And I have a feeling we passed that bar long, long ago. 

How can we get more details so this page can be updated?

**Comment by jensimmons** [comment]
Safari does support vertical writing modes for ruby.

We do not yet support `ruby-overhang` or `ruby-merge`. But given the fact they are not documented in MDN BCD, it's very doubtful any browser does. Firefox 38 certainly does not.

**Comment by Fyrd** [comment]
Yes, looking at the [test results here](https://w3c.github.io/i18n-tests/results/css-ruby) I agree Ruby CSS support is much more complex than is expressed in the support table.

Perhaps this table could be repurposed to refer to just the HTML elements and leave the CSS properties to their individual BCD tables?

**Comment by jensimmons** [comment]
That's what I was thinking too — if the HTML is supported, and changes writing modes (so — both horizontal and vertical ruby), then it's "fully supported".  

Then all the CSS properties for adjusting / refining ruby layout can be tracked one by one. 

This maps the rest of web tech. Do you support `<h1>` headlines? Great, done. Do you support these 97 CSS properties to refine how that headline looks? Let's list all of them, and provide support data one by one —especially since that list is always expanding.

**Comment by jensimmons** [comment]
The question then becomes how do we know when each browser gained "full support"? To answer that, we need to know what's required for "full". 

Supporting vertical writing modes seems required. Vertical typography is very common and popular in Japanese graphic design, and only supporting horizontal ruby seems limited. 

But what else?

**Comment by yisibl** [comment]
WebKit now supports `ruby-overhang`.
https://github.com/WebKit/WebKit/commit/1a18e1521bfc4b2affc1807d319fe3ecfdf9418f

**Comment by yisibl** [comment]
Also, the new Ruby specification is at https://w3c.github.io/html-ruby/

**Comment by fantasai** [comment]
Given that Firefox is marked as full support and others as partial, I suspect this is about support for ruby markup (`<ruby>`, `<rt>`, `<rb>`, `<rtc>`).  Gecko supports all of these tags, correctly handling pairing and multi-level annotations, whereas Blink and WebKit only support the first two.

**Comment by ar-an-ribe** [comment]
> Safari also does not support annotation spanning (ruby columns in general) yet. It's unclear if other browsers do.

In the case of a W3C HTML5 document with a horizontal layout in Firefox 45.9.0 ESR (released in April 2017), annotation spanning is supported. I haven’t tried a document with a vertical layout, though; my test document is tabular in nature, with Japanese readings and English annotations.

However, in a strictly compliant XHTML 1.1 version of my test document, Firefox 45.9.0 ESR doesn’t support the XHTML 1.1 method of annotation spanning via the `<rt>` element’s `rbspan` attribute; this attribute might only be found in XHTML 1.1. (I also had to add a stylesheet rule for the `<rbc>` element of XHTML 1.1, since no such rule was in the browser’s default stylesheet.)

The W3C HTML5 method of annotation spanning works in a XHTML 1.1 document in this browser, but such a document does not strictly comply with the XHTML 1.1 specification. It might be valid XHTML5, but I don’t know of a validator for XHTML5 documents.
